lib/src/executables/intl_message_migration.dart

```diff
@@ -0,0 +1,315 @@
+    }
+
+    return 0;
+  }
+
+  final FileSystem fs = const LocalFileSystem();
+  final dartPaths = dartFilesToMigrate().toList();
+  // Work around parts being unresolved if you resolve them before their libraries.
+  // TODO - reference analyzer issue for this once it's created
+  final packageRoots = dartPaths.map(findPackageRootFor).toSet().toList();
+  packageRoots.sort((packageA, packageB) =>
+      packageB.split('/').length - packageA.split('/').length);
+
+  Map<String, String> packageNameLookup = Map.fromIterable(
+    pubspecYamlPaths(),
+    key: (path) => findPackageRootFor(path).split('/').last,
+    value: (path) {
+      final pubspec = fs.file(path);
+      List<String> pubspecLines = pubspec.readAsLinesSync();
+      String nameLine =
+          pubspecLines.firstWhere((line) => line.startsWith('name'));
+      var nameLineParts = nameLine.split(':');
+      return nameLineParts[1].trim();
+    },
+  );
+
+  final processedPackages = Set<String>();
+  await pubGetForAllPackageRoots(dartPaths);
+
+  exitCode = await runInteractiveCodemod(
+    [],
+    (_) async* {},
+    args: codemodArgs,
+    additionalHelpOutput: parser.usage,
+  );
+  if (exitCode != 0) return;
+  print('^ Ignore the "codemod found no files" warning above for now.');
+
+  for (String package in packageRoots) {
+    _log.info('Starting migration for $package');
+
+    final packageRoot = package.split('/').last;
+    final packageName = packageNameLookup[packageRoot] ?? 'fix_me_bad_name';
+    _log.info('Starting migration for $packageName');
+    final packageDartPath =
+        dartFilesToMigrateForPackage(package, processedPackages).toList();
+    sortPartsLast(packageDartPath);
+
+    final File outputFile = fs.currentDirectory
+        .childFile('${package}/lib/src/intl/${packageName}_intl.dart');
+    final bool existingOutputFile = outputFile.existsSync();
+
+    final className = toClassName('${packageName}');
+    if (!existingOutputFile) {
+      outputFile.createSync(recursive: true);
+      outputFile.writeAsStringSync(
+          "import 'package:intl/intl.dart';\n\n//ignore: avoid_classes_with_only_static_members\nclass $className {\n");
+    } else {
+      final List<String> lines = outputFile.readAsLinesSync();
+      lines.removeLast();
+      String outputContent = lines.join('\n');
+      outputFile.writeAsStringSync(outputContent);
+    }
+
+    final intlPropMigrator = IntlMigrator(className, outputFile);
+    final displayNameMigrator = ConfigsMigrator(className, outputFile);
+    final importMigrator =
+        (FileContext context) => intlImporter(context, packageName, className);
+
+
+    exitCode = await runCodemodSequences(packageDartPath, [
+      [intlPropMigrator],
+      [displayNameMigrator],
+      [importMigrator]
+    ]);
+
+    processedPackages.add(package);
+    if (exitCode != 0) {
+      outputFile.deleteSync();
+    } else {
+      List<String> lines = outputFile.readAsLinesSync();
+      if (lines.length == 4) {
```

Comment:
Where does this 4 come from?

Reply:
```Dart
import "package:intl/intl.dart";

//ignore: avoid_classes_with_only_static_members
class CLASSNAME {
```

Reply:
I was doing this via a "wasModified" flag to start, but that wasn't idempotent as a subsequent run would delete the file.

Reply:
This seems a bit brittle. What about storing that code as a variable and checking whether the file contains only that code?